concourse/pipelines/create-deployer.yml

```diff
@@ -280,6 +287,28 @@ jobs:
         params:
           file: generated-concourse-cert/concourse-cert.tar.gz
 
+    - task: generate-concourse-ssh-key
+      config:
+        image: docker:///governmentpaas/curl-ssl
+        inputs:
+        - name: paas-cf
+        outputs:
+        - name: concourse-ssh-key
+        params:
+          AWS_DEFAULT_REGION: {{aws_region}}
+        run:
+          path: sh
+          args:
+          - -e
+          - -c
+          - |
+            apk add --update openssh
+            cd concourse-ssh-key
+            ssh-keygen -t rsa -b 4096 -f generated_concourse_id_rsa -N ''
+            ../paas-cf/concourse/scripts/s3init.sh {{state_bucket}} concourse_id_rsa generated_concourse_id_rsa
+            ../paas-cf/concourse/scripts/s3init.sh {{state_bucket}} concourse_id_rsa.pub generated_concourse_id_rsa.pub
```

Comment:
> Can we use put to upload the generated key instead of s3init?

That gives us the bootstrapping problem. `s3init` is used here to only upload the key if it's not already in the state bucket.  Using a put would result in rotating the key on every deploy (unless we add some more logic around it).